Source/Events.Processing/EventHandlers/EventHandlersService.cs

```diff
@@ -196,37 +196,44 @@ public class EventHandlersService : EventHandlersBase
                 cancellationToken).ConfigureAwait(false);
             if (!tryStartEventHandler.Success)
             {
-                internalCancellationTokenSource.Cancel();
+                linkedTokenSource.Cancel();
                 if (tryStartEventHandler.HasException)
                 {
                     var exception = tryStartEventHandler.Exception;
-                    _logger.Debug(exception, "An error occurred while starting Event Handler: '{eventHandlerId}' in Scope: {scopeId}", eventHandlerId, scopeId);
+                    _logger.Debug(exception, "An error occurred while starting Event Handler: '{EventHandlerId}' in Scope: {ScopeId}", eventHandlerId, scopeId);
                     ExceptionDispatchInfo.Capture(exception.InnerException).Throw();
                 }
                 else
                 {
-                    _logger.Debug("Could not start Event Handler: '{eventHandlerId}' in Scope: {scopeId}", eventHandlerId, scopeId);
+                    _logger.Debug("Could not start Event Handler: '{EventHandlerId}' in Scope: {ScopeId}", eventHandlerId, scopeId);
                     return;
                 }
             }
 
             var tasks = tryStartEventHandler.Result;
             var anyTask = await Task.WhenAny(tasks).ConfigureAwait(false);
+
             if (TryGetException(tasks, out var ex))
             {
-                internalCancellationTokenSource.Cancel();
-                _logger.Warning(ex, "An error occurred while processing Event Handler: '{eventHandlerId}' in Scope: '{scopeId}'", eventHandlerId, scopeId);
+                _logger.Warning(ex, "An error occurred while processing Event Handler: '{EventHandlerId}' in Scope: '{ScopeId}'", eventHandlerId, scopeId);
+                linkedTokenSource.Cancel();
                 await Task.WhenAll(tasks).ConfigureAwait(false);
                 ExceptionDispatchInfo.Capture(ex.InnerException).Throw();
             }
+            else
+            {
+                linkedTokenSource.Cancel();
+                await Task.WhenAll(tasks).ConfigureAwait(false);
+            }
```

Comment:
Nice catch. To my eyes the missing piece here was the linkedTokenSource.Cancel() call beneath the if-block. And since we throw an exception at the end of the if-block we don't actually need the else-block.